### ISSUE TITLE
API Rest generates wrong endpoint

### ISSUE THREAD

**Issue by Albert-Gao** [aws-amplify/amplify-cli#4245]
Amplify 4.19.0

After creating, it gives me `ANY` in API Gateway

So I did an `amplify API update`

![image](https://user-images.githubusercontent.com/18282328/81808578-04f2fe00-9574-11ea-88d9-d3d8b4317b4e.png)

But still, there is no POST, but any

![image](https://user-images.githubusercontent.com/18282328/81808683-2eac2500-9574-11ea-8115-fb72903f3bf9.png)

Anything I am missing here?

Does it support to create a POST endpoint?

Or I need to filter it by myself in my lambda function?

**Comment by Albert-Gao** [comment]
I removed the API + Function, and re-created, but still there is no `POST` endpoint in the API Gateway

This is the creating procedure:

![image](https://user-images.githubusercontent.com/18282328/81900263-90ba6800-9610-11ea-82c6-a7a4e8d3bb6c.png)

If I get it right, `create` means `POST` here, right?

**Comment by Albert-Gao** [comment]
Found this issue

https://github.com/aws-amplify/amplify-cli/issues/1232

So it is a missing feature, actually.

Why that issue is closed...
Is this on the roadmap?

**Comment by Albert-Gao** [comment]
I am able to add it by editing the cloudformation template json

But still have some problems

![image](https://user-images.githubusercontent.com/18282328/81909827-3f65a500-961f-11ea-8765-2eb1d5c236cb.png)

I am able to attach the method to the top level resource but not the paths it creates, I tried, but it gives me a 400 error

So, this is how I added the METHOD resources:

```json
{
    "createCompanyMethod": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "RestApiId": {
          "Ref": "createCompany"
        },
        "ResourceId": {
          "Fn::GetAtt": ["createCompany", "RootResourceId"]
        },
        "HttpMethod": "POST",
        "AuthorizationType": "NONE",
        "Integration": {
          "Type": "MOCK"
        }
      }
    }
}
```

When I am trying to add method to the paths like this:

I read the Open API example, added a nested Http method object

![image](https://user-images.githubusercontent.com/18282328/81910134-a84d1d00-961f-11ea-81cc-fc16d2f94d31.png)

This is the error when doing the depolyment

![image](https://user-images.githubusercontent.com/18282328/81910038-910e2f80-961f-11ea-8340-ea46313759d5.png)

Anything I am missing here?  @kaustavghosh06 Sorry to tag you, but I saw you in that issue #1232

**Comment by edwardfoyle** [comment]
@Albert-Gao you are correct that this configuration is not supported through the Amplify CLI. The REST service configured by the CLI is primarily geared towards hosting an ExpressJS server running in lambda that will multiplex all requests rather than separate lambdas for each HTTP verb. I'm going to tag this as an enhancement.

Also regarding your OpenAPI spec, I don't think the "options" block should be under "post". OPTIONS is an HTTP verb the same as POST or GET. See https://swagger.io/specification/#path-item-object

**Comment by WholeEggz** [comment]
@Albert-Gao I see that you are looking at the 'resources' tab (left column on api gateway dashboard). Check 'stages' tab below - there you should see all methods

**Comment by glynjackson** [comment]
**Can we have an update on this, please?** 

The original feature was requested in **2019** and should not have been closed https://github.com/aws-amplify/amplify-cli/issues/1232 In 2021, and they're still doesn't appear to be a way to define a Rest API method; the default is "ANY",

- Currently, my Lambda functions are triggered on every method, meaning I could be **incurring additional costs**!  
- Editing the cloud-formation script runs the **risk of it being overwritten**.
- Not everyone uses ExpressJS, and on some endpoint, we don't want to support every method.
- Amplify could expose us to additional security risks by not supporting this feature.

Workarounds would be appreciated. I have only been using Amplify for a few months and not built-up the skills to know if I've missed something or submit a PR.

Thanks
